metabox/metabox/core/machine.py

```diff
@@ -28,9 +29,8 @@
 from metabox.core.lxd_execute import interactive_execute
 from metabox.core.lxd_execute import run_or_raise
 
-
 class MachineConfig:
-
+    NAME_VALID_CHAR_RE = re.compile(r"[\w\d]+")
```

Comment:
I will add a docstr on top of it, the idea is that this regex is used to select all valid chars in the name to create a "pretty" and valid name for the revision. By pretty I mean dash separated words or numbers

Reply:
The conclusions I'm drawing from your explanation:
- the core concept is "revision", not name
- there are more than one char that's allowed

Conclusion on the regex itself: both `\w` and `\d` are not even characters, they are classes.

One last thing, the `\d` is redundant as it's included in `\w`

Final thoughts:
- this regex is not nedded _at all_ imho
- if you as an author made those mistakes it will be even worse for the reader.

Please kill it.

Reply:
I don't get you, this is used to select all valid chars in the provided revision name for instance in `origin/main` -> `origin` `main`. I wanted to avoid `/` in lxd machine names because that is not a valid char AND I do not know, but I suspect, there are more that I'm not thinking about. So instead of thinking about every possible invalid char in the revision name, I have created this tiny regex that matches a sane subset of them. About the`\d`, didnt recall that, I will remove it.

Reply:
There is only one revision we care about - `main`. Let's use that name. The thing on the other side is just the current source. I don't see the reason why we need to deal with any more revision names.

As a complete side note (that shouldn't be applied here) - I think git is already very restrictive when it comes to branch names

Reply:
no, we care about `origin/main` and it contains a `/`. And we may care about any other revision in the future (a tag? for example), this is way more powerful with so little work.